sceptre/cli.py

```diff
@@ -574,6 +576,170 @@ def get_stack_policy(ctx, environment, stack):
+    :type config_dir: str
+    :param path: The directory path to the environment folder.
+    :type path: str
+    """
+    # Create full path to environment
+    folder_path = os.path.join(config_dir, new_path)
+    environment_exists = False
+
+    # Make folders for the anvironment
+    try:
+        os.makedirs(folder_path)
+    except OSError as e:
+        # Check if environment folder already exists
+        if e.errno == errno.EEXIST:
+            environment_exists = True
+        else:
+            raise
+
+    init_config_msg = 'Do you want initialise config.yaml?'
+
+    if environment_exists:
+        init_config_msg = 'Environment path exists. ' + init_config_msg
+
+    if click.confirm(init_config_msg):
+        _create_config_file(config_dir, folder_path)
+
+
+def _get_nested_config(config_dir, path):
+    """
+    Collects nested config from between `config_dir` and `path`. Config at
+    lower level as greater precedence.
+
+    :param config_dir: The directory path to the top-level config folder.
+    :type config_dir: str
+    :param path: The directory path to the environment folder.
+    :type path: str
+    :returns: The nested config.
+    :rtype: dict
+    """
+    config = {}
+    for root, _, files in os.walk(config_dir):
+        # Check that folder is within the final environment path
+        if path.startswith(root) and "config.yaml" in files:
+            config_path = os.path.join(root, "config.yaml")
+            with open(config_path) as config_file:
+                config.update(yaml.safe_load(config_file))
+    return config
+
+
+def _create_config_file(config_dir, path, defaults=None):
+    """
+    Creates a `config.yaml` file in the given path. The user is asked for
+    values for requried properties. Defaults are suggested with values in
+    `defaults` and then vaules found in parent `config.yaml` files. If
+    properties and their values are the same as in parent `config.yaml`, then
+    they are not included. No file is produced if require values are satisfied
+    by parent `config.yaml` files.
+
+    :param config_dir: The directory path to the top-level config folder.
+    :type config_dir: str
+    :param path: The directory path to the environment folder.
+    :type path: str
+    :param defaults: Defaults to present to the user for config.
+    :type defaults: dict
+    """
+    config = dict.fromkeys(ENVIRONMENT_CONFIG_ATTRIBUTES.required, "")
+    nested_config = _get_nested_config(config_dir, path)
+
+    # Add nested config as defaults
+    config.update(nested_config)
+
+    if defaults:
+        config.update(defaults)
+
+    # Ask for new values
+    for key, value in config.items():
+        config[key] = click.prompt(
+            'Please enter a {0}'.format(key), default=value
+        )
+
+    # Remove nested values that are the same
+    config = {k: v for k, v in config.items() if nested_config.get(k) != v}
+
```

Comment:
A bit nit-picky, but it seems like the expressions containing nested_config could (and should) be grouped together.

Reply:
Ah not sure what you mean - could you explain further?

Reply:
It may be easier to explain in person (so I can point to code and use words like "this", "that", and "there"). It's not a huge deal either way.